lib/handleRequest.js

```diff
@@ -89,28 +89,29 @@ function preValidationCallback (err, request, reply) {
     return
   }
 
-  const result = validateSchema(reply[kRouteContext], request)
-  if (result) {
-    if (reply[kRouteContext].attachValidation === false) {
-      reply.send(result)
-      return
-    }
+  validateSchema(reply[kRouteContext], request, function cb (err) {
```

Comment:
If it's sync validation, this callback should not be allocated at all.

Reply:
I can rewrite it as this:
<img width="519" alt="image" src="https://github.com/fastify/fastify/assets/11404065/86bd187b-e3b1-499a-975d-792cb4e7d382">

But I need at least one function to process the callback in sync or async manner

Do you have something else in mind?

Reply:
Friendly ping - without adding a single callback, the code becomes very messy

When the execution is sync, the callback is executed immediately, there should not be a loop re-run

Reply:
We would need to benchmark this very carefully because adding a callback adds overhead and the reason to not support this from the beginning was said overhead. The only thing we have on our side is the fact that V8 improved.

Another option is to just duplicate code if it's sync or async.

Reply:
I have pushed the no-callback solution - the coverage must be fixed, but if benchmarks would be good, would that code fine?